### PR TITLE
⚠️ Replace SecurityGroupParam with SecurityGroupFilter

### LINKED ISSUE #1548: Remove SecurityGroupParam
/kind feature

SecurityGroupParam contains a SecurityGroupFilter object which already contains both of the other fields of SecurityGroupParam. We should replace SecurityGroupParam with SecurityGroupFilter.

SecurityGroupParam is used in PortOpts.SecurityGroupFilters and OpenStackMachineSpec.SecurityGroups.

### PR DESCRIPTION
**Pull request by mdbooth**
SecurityGroupParam was:
```golang
type SecurityGroupParam struct {
       // Security Group UID
       UUID string `json:"uuid,omitempty"`
       // Security Group name
       Name string `json:"name,omitempty"`
       // Filters used to query security groups in openstack
       Filter SecurityGroupFilter `json:"filter,omitempty"`
}
```

However, the SecurityGroupFilter is contains also has ID and Name fields, making this wrapper redundant.

This is a breaking change due the removal of nesting of `filter` and also because the ID field is now called `ID` instead of `UUID`.

Fixes #1548 

- [x] update docs

/hold

### CODE REVIEW

File: docs/book/src/topics/crd-changes/v1alpha6-to-v1alpha7.md [hunk 1]
```diff
@@ -112,6 +112,41 @@ to set the `accessIPv4` field on Nova servers. This feature was not widely
...
+```
+becomes:
+```yaml
+securityGroups:
+  tags: my-security-group-tag
+```
+
+#### Type change to Port SecurityGroupFilters
+
+The same change is made to `securityGroupFilters` in `ports` as is made to `securityGroups` in the machine spec.
```
Comment (seanschneeweiss) [inline docs/book/src/topics/crd-changes/v1alpha6-to-v1alpha7.md:148]: Shall we link to the above heading?

Comment (mdbooth) [reply docs/book/src/topics/crd-changes/v1alpha6-to-v1alpha7.md:148]: If I can work out how!

**Review by lentzi90 — COMMENTED** [review]
Nice!
/lgtm

**Comment by jichenjc** [comment]
/lgtm

maybe squash the commit then we should be good to go

**Pull request merged**